packages/core/src/client/hmr/index.ts

```diff
@@ -216,3 +172,110 @@ function tryApplyUpdates() {
     );
   }
 }
+
+const MAX_RETRIES = 20;
```

Comment:
Some large app may need more time to restart the dev server, so MAX_RETRIES = 100 can be better

```suggestion
const MAX_RETRIES = 100;
```